editor/project_manager.cpp

```diff
@@ -107,21 +107,25 @@ void ProjectDialog::_set_message(const String &p_msg, MessageType p_type, InputT
 	}
 }
 
+static bool is_zip_file(Ref<DirAccess> p_d, const String &p_path) {
+	return p_path.ends_with(".zip") && p_d->file_exists(p_path);
+}
+
 String ProjectDialog::_test_path() {
 	Ref<DirAccess> d = DirAccess::create(DirAccess::ACCESS_FILESYSTEM);
+	const String base_path = project_path->get_text();
 	String valid_path, valid_install_path;
-	if (d->change_dir(project_path->get_text()) == OK) {
-		valid_path = project_path->get_text();
-	} else if (d->change_dir(project_path->get_text().strip_edges()) == OK) {
-		valid_path = project_path->get_text().strip_edges();
-	} else if (project_path->get_text().ends_with(".zip")) {
-		if (d->file_exists(project_path->get_text())) {
-			valid_path = project_path->get_text();
-		}
-	} else if (project_path->get_text().strip_edges().ends_with(".zip")) {
-		if (d->file_exists(project_path->get_text().strip_edges())) {
-			valid_path = project_path->get_text().strip_edges();
-		}
+	bool is_zip = false;
+	if (d->change_dir(base_path) == OK) {
+		valid_path = base_path;
+	} else if (is_zip_file(d, base_path)) {
```

Comment:
Reordering here in the weird case where there's a file matching but not a directory, unsure if this will ever happen, but it can't hurt